compiler/rustc_lint/src/drop_forget_useless.rs

```diff
@@ -147,19 +147,31 @@ impl<'tcx> LateLintPass<'tcx> for DropForgetUseless {
             let arg_ty = cx.typeck_results().expr_ty(arg);
             let is_copy = arg_ty.is_copy_modulo_regions(cx.tcx, cx.param_env);
             let drop_is_single_call_in_arm = is_single_call_in_arm(cx, arg, expr);
+            let sugg = if let Some((_, node)) = cx.tcx.hir().parent_iter(expr.hir_id).nth(0)
```

Comment:
Accessing the `hir()` query is not free and there is no reason pre-compute the suggestion when we are not using it. Could you transform this into a closure.
```suggestion
            let sugg = || if let Some((_, node)) = cx.tcx.hir().parent_iter(expr.hir_id).nth(0)
```
and then call it where necessary.